src/features/counterfactual/utils.ts

```diff
@@ -35,3 +41,46 @@ export const getUndeployedSafeInfo = (undeployedSafe: PredictedSafeProps, addres
     deployed: false,
   })
 }
+
+export const dispatchTxExecutionAndDeploySafe = async (
+  safeTx: SafeTransaction,
+  txOptions: TransactionOptions,
+  onboard: OnboardAPI,
+  chainId: SafeInfo['chainId'],
+) => {
+  const sdkUnchecked = await getUncheckedSafeSDK(onboard, chainId)
+
+  let result: ContractTransactionResponse | undefined
+  try {
+    const wallet = await assertWalletChain(onboard, chainId)
+
+    const signedTx = await sdkUnchecked.signTransaction(safeTx)
+    const deploymentTx = await sdkUnchecked.wrapSafeTransactionIntoDeploymentBatch(signedTx, txOptions)
+
+    const provider = createWeb3(wallet.provider)
+    const signer = await provider.getSigner()
+
+    // @ts-ignore TODO: Check if the other type also works
+    result = await signer.sendTransaction(deploymentTx)
```

Comment:
Some of these calls could be parallelized as well, not sure how fast it is already.

Reply:
We could run the `wallet` and `deploymentTx` call in parallel but I think its already fast currently